spock-specs/src/test/groovy/org/spockframework/smoke/WithBlocks.groovy

```diff
@@ -65,6 +65,43 @@ class WithBlocks extends Specification {
     e.message.contains("Person")
   }
 
+  def "method condition is invoked on closure but not on the spec"() {
+
+    def map = [ 'value1' : 1, 'value2' : 2]
+
+    expect:
+    with(map) {
+      size() == 2
+      containsKey('value2')
+    }
+  }
+
+  def "nested method conditions are invoked on closure but not on the spec"() {
```

Comment:
I would split it into 3 tests (to have a clear picture what failed in a case of regression).